### PR TITLE
[Bye bye Spree] Make OFN independent of all spree spec factories

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?
Relates to #4826 
Move all necessary factories from spree_core to ofn.

#### What should we test?
A green build is enough here, only test code changed.

#### Release notes
Changelog Category: Changed
Brought test support code from spree so that we can make OFN independent of Spree.

### CODE REVIEW

File: spec/factories.rb [hunk 1]
```diff
@@ -1,22 +1,10 @@
 require 'ffaker'
-require 'spree/testing_support/factories'
```
Comment (luisramos0) [inline spec/factories.rb:2]: this is the magic of this PR: no spree factories used.

**Review by Matt-Yorkley — APPROVED** [review]

File: spec/factories/variant_factory.rb [hunk 2]
```diff
@@ -1,46 +1,66 @@
...
-        producer { product.supplier }
-        coordinator { create(:distributor_enterprise) }
-        distributor { create(:distributor_enterprise) }
-        incoming_exchange_fees { [] }
-        outgoing_exchange_fees { [] }
+        on_demand { false }
+        on_hand { 5 }
       end
 
+      # on_hand 5
```
Comment (sauloperez) [inline spec/factories/variant_factory.rb:25]: :fire:

File: spec/models/spree/stock_item_spec.rb [hunk 3]
```diff
@@ -3,7 +3,15 @@
...
 RSpec.describe Spree::StockItem do
-  let(:stock_location) { create(:stock_location_with_items) }
+  let(:stock_location) { create(:stock_location) }
+
+  before do
+    product_1 = create(:product)
+    product_2 = create(:product)
+
+    stock_location.stock_items.where(:variant_id => product_1.master.id).first.adjust_count_on_hand(10)
+    stock_location.stock_items.where(:variant_id => product_2.master.id).first.adjust_count_on_hand(20)
```
Comment (sauloperez) [inline spec/models/spree/stock_item_spec.rb:13]: we can replace the hash rocket while we are at it :pray:

**Review by sauloperez — CHANGES_REQUESTED** [review]
:clap:

File: spec/factories/variant_factory.rb (hunk 2 above)
Comment (luisramos0) [reply spec/factories/variant_factory.rb:25]: :+1:

File: spec/models/spree/stock_item_spec.rb (hunk 3 above)
Comment (luisramos0) [reply spec/models/spree/stock_item_spec.rb:13]: well spotted :+1:

**Comment by luisramos0** [comment]
:+1: I ended up running rubocop -a on spec/factories (last commit). Let's see if the build is ok.

**Review by sauloperez — APPROVED** [review]

**Pull request merged**